### PR TITLE
[OUR415-314] Fixes eligibility and category lists in Browse Experience

### PR DESCRIPTION
**Pull request by rosschapman**
Also: 
- Updates recently added `BrowseRefinementList` test, proving that tests are useful!

### CODE REVIEW

File: app/components/ui/Navigation/DropdownMenu.module.scss [hunk 1]
```diff
@@ -92,11 +90,12 @@
...
   position: relative;
-  min-width: 160px;
+
 
   .navigationSubMenu {
-    top: 50px;
-    left: 0;
+    @include r_max($break-tablet-s) {
+      width: calc(100vw - $general-spacing-xs);
+    }
```
Comment (rosschapman) [inline app/components/ui/Navigation/DropdownMenu.module.scss:98]: **note:** This expands the top level category dropdown full width. Forgot to implement it before.

File: app/components/search/Sidebar/Sidebar.module.scss [hunk 2]
```diff
@@ -38,6 +38,7 @@
     align-self: flex-start;
     border-right: 0;
     background-color: #fff;
+    z-index: $z-index-filters-menu;
```
Comment (rosschapman) [inline app/components/search/Sidebar/Sidebar.module.scss:41]: **note:** Discovered a bug where the filters were not on top.

File: app/components/ui/Navigation/DropdownMenu.module.scss (hunk 1 above)
Comment (adriencyberspace) [reply app/components/ui/Navigation/DropdownMenu.module.scss:98]: Just wondering - Is there a reason for choosing general-spacing-xs? Or for adding that only on the category dropdown rather the nav dropdown too?

File: app/components/search/Sidebar/Sidebar.module.scss (hunk 2 above)
Comment (adriencyberspace) [reply app/components/search/Sidebar/Sidebar.module.scss:41]: Speaking of z-indexes. The category dropdown appears to have a higher z-index than the navigation bar. If that would be a quick fix, might make sense to put it in this PR

image

**Review by adriencyberspace — APPROVED** [review]
LGTM

File: test/helpers/createRandomCategories.ts [hunk 3]
```diff
@@ -0,0 +1,20 @@
+import { faker } from "@faker-js/faker";
```
Comment (adriencyberspace) [inline test/helpers/createRandomCategories.ts:1]: **praise:** cool 😎

File: app/components/ui/Navigation/DropdownMenu.module.scss (hunk 1 above)
Comment (rosschapman) [reply app/components/ui/Navigation/DropdownMenu.module.scss:98]: Only that I discovered this variable defined the correct spacing.

File: app/components/search/Sidebar/Sidebar.module.scss (hunk 2 above)
Comment (rosschapman) [reply app/components/search/Sidebar/Sidebar.module.scss:41]: What browser/resolution is that? I don't even know how to get in that state 😅

**Pull request merged**